src/main/java/org/folio/circulation/services/FeeFineService.java

```diff
@@ -1,58 +1,165 @@
 package org.folio.circulation.services;
 
 import static java.util.concurrent.CompletableFuture.allOf;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.services.feefine.AccountRefundProcessors.getProcessor;
+import static org.folio.circulation.support.Result.combined;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
 
 import java.util.Collection;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.folio.circulation.domain.Account;
 import org.folio.circulation.domain.AccountRepository;
-import org.folio.circulation.domain.FeeFineAccountAndAction;
 import org.folio.circulation.domain.FeeFineActionRepository;
-import org.folio.circulation.domain.representations.AccountStorageRepresentation;
-import org.folio.circulation.domain.representations.FeeFineActionStorageRepresentation;
+import org.folio.circulation.domain.ServicePoint;
+import org.folio.circulation.domain.ServicePointRepository;
+import org.folio.circulation.domain.User;
+import org.folio.circulation.domain.UserRepository;
+import org.folio.circulation.domain.representations.StoredAccount;
+import org.folio.circulation.domain.representations.StoredFeeFineAction;
+import org.folio.circulation.services.feefine.AccountRefundContext;
+import org.folio.circulation.services.feefine.AccountRefundProcessor;
+import org.folio.circulation.services.support.AccountCreation;
+import org.folio.circulation.services.support.AccountRefund;
 import org.folio.circulation.support.Clients;
 import org.folio.circulation.support.Result;
 import org.folio.circulation.support.results.CommonFailures;
 
+import io.vertx.core.json.JsonObject;
+
 public class FeeFineService {
   private final AccountRepository accountRepository;
   private final FeeFineActionRepository feeFineActionRepository;
+  private final UserRepository userRepository;
+  private final ServicePointRepository servicePointRepository;
 
   public FeeFineService(Clients clients) {
     this.accountRepository = new AccountRepository(clients);
     this.feeFineActionRepository = new FeeFineActionRepository(clients);
+    this.userRepository = new UserRepository(clients);
+    this.servicePointRepository = new ServicePointRepository(clients);
   }
 
   public CompletableFuture<Result<Void>> createAccountsAndActions(
-    Collection<FeeFineAccountAndAction> accountAndActions) {
+    Collection<AccountCreation> accountAndActions) {
```

Comment:
This class holds required data to create an account and action for it, e.g.:
* loan;
* item;
* service point;
* staff user id.
* etc.

Reply:
Is it intended to represent an instruction to issue a fee?

Reply:
Yes, it is.

Reply:
Could this be named after that intent, e.g. `CreateAccount`, `CreateAccountInstruction`, `CreateAccountCommand` (I'm hesitant to suggest `CreateAccountRequest` because that term is overloading in mod-circulation)?

Reply:
Renamed to `CreateAccountCommand`